src/internet_identity/src/storage/anchor.rs

```diff
@@ -164,6 +164,25 @@ impl Anchor {
         &self.devices
     }
 
+    /// Returns matching [DeviceCredential]s.
+    pub fn get_device_credentials<P>(&self, filter: P) -> Vec<DeviceCredential>
```

Comment:
I think it may be simpler to do everything in `get_query_credentials`, I find it a little odd that this module ("storage") needs to know about extracting credentials

Reply:
I originally had it that way, but without `partition` the code was very cumbersome. But since I now know it exists, I have removed this method again.